codalab/lib/bundle_cli.py

```diff
@@ -812,6 +812,31 @@ def do_search_command(self, argv, parser):
                 client.add_worksheet_item(worksheet_uuid, worksheet_util.bundle_item(bundle_uuid))
             worksheet_info = client.get_worksheet_info(worksheet_uuid, False)
             print 'Added %d bundles to %s' % (len(bundle_uuids), self.worksheet_str(worksheet_info))
+        return self.create_structured_info_map([('refs', reference_map),])
+
+    def create_structured_info_map(self, structured_info_list):
+        '''
+        Return dict of info dicts (eg. bundle/worksheet reference_map) containing
+        information associated to bundles/worksheets. cl wls, ls, etc. show uuids
+        which are too short. This dict contains additional information that is
+        needed to recover URL on the client side.
+        '''
+        return { k: v for k, v in structured_info_list}
+
+    def create_reference_map(self, info_type, info_list):
+        '''
+        Return dict of dicts containing name, uuid and type for each bundle/worksheet
+        in the info_list. This information is needed to recover URL on the cient side.
+        '''
+        if len(info_list) == 0:
+            return None
```

Comment:
This was mostly a design decision. An empty dictionary could be returned but I decided to go with a None since that can be easily be checked wherever this method is called and one does not have find the length of dictionary to know that there is no information present.